graylog2-server/src/main/java/org/graylog2/shared/security/ShiroSecurityContextFilter.java

```diff
@@ -39,17 +41,21 @@
 @Priority(Priorities.AUTHENTICATION)
 public class ShiroSecurityContextFilter implements ContainerRequestFilter {
     private final DefaultSecurityManager securityManager;
+    private Provider<Request> grizzlyRequestProvider;
 
     @Inject
-    public ShiroSecurityContextFilter(DefaultSecurityManager securityManager) {
+    public ShiroSecurityContextFilter(DefaultSecurityManager securityManager, Provider<Request> grizzlyRequestProvider) {
         this.securityManager = requireNonNull(securityManager);
+        this.grizzlyRequestProvider = grizzlyRequestProvider;
     }
 
     @Override
     public void filter(ContainerRequestContext requestContext) throws IOException {
         final boolean secure = requestContext.getSecurityContext().isSecure();
         final MultivaluedMap<String, String> headers = requestContext.getHeaders();
-        final String host = headers.getFirst(HttpHeaders.HOST);
+        final Request grizzlyRequest = grizzlyRequestProvider.get();
+
+        final String host = grizzlyRequest.getRemoteAddr();
```

Comment:
This won't work with reverse proxies in front of Graylog. We should at least support the "standard" headers like `X-Forwarded-For` (also see `RestTools#getRemoteAddrFromRequest()`).